server/controllers/user.js

```diff
@@ -20,12 +20,50 @@ exports.searchUsers = asyncHandler(async (req, res, next) => {
     throw new Error("No users found in search");
   }
 
-  return res.status(200).json({ users: users });
+  res.status(200).json({ users: users });
+});
+
+exports.setProfilePicture = asyncHandler(async (req, res) => {
+  if (!req.picture) return res.status(404).send("Picture not found");
+
+  const picture = { url: req.picture.url };
+
+  const user = await User.findOneAndUpdate(
+    { _id: req.user.id },
+    { $set: { picture: picture } },
+    { new: true }
+  );
+
+  if (user) return res.status(200).send({ picture });
+
+  return res.status(400).send({ error: "Could not retrieve user!" });
+});
```

Comment:
Use AsyncHandler's `throw new Error(...)` functionality instead of `status(404).send(...)`